balances-adjustment/src/main/java/com/powsybl/balances_adjustment/util/InjectionBasedVoltageLevelsArea.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.balances_adjustment.util;
+
+import com.powsybl.iidm.network.Injection;
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author Joris Mancini <joris.mancini_externe at rte-france.com>
+ */
+public class InjectionBasedVoltageLevelsArea extends AbstractInjectionBasedArea {
```

Comment:
Could you add a test that uses this class please ?  (for example with a test with `<VoltageLevelsAreaFactory>.create(network, true)`)